### PR TITLE
zb: pick client IPs from a pool, closes #472

### LINKED ISSUE #472: zb: pick client IPs from a pool
**Is your feature request related to a problem? Please describe.**

Currently, zb is not aware of src ips to pick from to setup the clients. This becomes important for a load-balancer use case wherein we want to test/simulate clients with different src ips.

**Describe the solution you'd like**

zb --src-ips 127.0.0.0/8
(OR)
zb --src-ips 127.0.0.1,127.0.0.2...

so that zb can randomly pick a src ip and in golang you can set the src ip of the http client like so.

```
OKAddr := "<change here>" // local IP address to use

         OKAddress, _ := net.ResolveTCPAddr("tcp", OKAddr)

         transport := &http.Transport{
                 Proxy: http.ProxyFromEnvironment,
                 Dial: (&net.Dialer{
                         Timeout:   30 * time.Second,
                         KeepAlive: 30 * time.Second,
                         OKAddr: OKAddress}).Dial, TLSHandshakeTimeout: 10 * time.Second}

         client := &http.Client{
                 Transport: transport,
         }
```

### PR DESCRIPTION
**Pull request by peusebiu**
Signed-off-by: Petu Eusebiu <peusebiu@cisco.com>

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by peusebiu** [comment]
Should I also add an argument for CIDR ? --src-cidr 127.0.0.1/8 and pick ips from there?

**Comment by peusebiu** [comment]
added -src-cidr argument

**Comment by rchincha** [comment]
@peusebiu can you just show us zot logs of some zb runs? I am looking for different client IPs in the zot logs.

**Comment by peusebiu** [comment]
> @peusebiu can you just show us zot logs of some zb runs? I am looking for different client IPs in the zot logs.

```
{"level":"info","module":"http","clientIP":"127.0.2.201:59979","method":"HEAD","path":"/v2/6de5f5ab-cad3-11ec-840b-460801757087/blobs/sha256:315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8","statusCode":200,"latency":"77.4µs","bodySize":0,"headers":{"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":565,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:17.8845936+03:00","message":"HTTP API"}
{"level":"debug","blob path":"/tmp/zot/6de5f967-cad3-11ec-840b-460801757087/blobs/sha256/cdc832d10daa3252dc66ba0744911d183177763a328f4b076f8ba664eb3e316b","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1260","time":"2022-05-03T14:23:17.8848461+03:00","caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1260","message":"blob path found"}
{"level":"info","module":"http","clientIP":"127.0.0.15:56249","method":"HEAD","path":"/v2/6de5f967-cad3-11ec-840b-460801757087/blobs/sha256:cdc832d10daa3252dc66ba0744911d183177763a328f4b076f8ba664eb3e316b","statusCode":200,"latency":"77.6µs","bodySize":0,"headers":{"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":566,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:17.8848852+03:00","message":"HTTP API"}
{"level":"info","module":"http","clientIP":"127.0.0.15:56249","method":"GET","path":"/v2/6de5f967-cad3-11ec-840b-460801757087/blobs/sha256:cdc832d10daa3252dc66ba0744911d183177763a328f4b076f8ba664eb3e316b","statusCode":200,"latency":"37.3µs","bodySize":144,"headers":{"Accept-Encoding":["gzip"],"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":566,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:17.8853462+03:00","message":"HTTP API"}
{"level":"debug","blob path":"/tmp/zot/6de5f967-cad3-11ec-840b-460801757087/blobs/sha256/315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1260","time":"2022-05-03T14:23:17.8857592+03:00","caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1260","message":"blob path found"}
{"level":"info","module":"http","clientIP":"127.0.0.15:56249","method":"HEAD","path":"/v2/6de5f967-cad3-11ec-840b-460801757087/blobs/sha256:315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8","statusCode":200,"latency":"91.4µs","bodySize":0,"headers":{"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":566,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:17.8859054+03:00","message":"HTTP API"}
{"level":"info","module":"http","clientIP":"127.0.2.201:59979","method":"GET","path":"/v2/6de5f5ab-cad3-11ec-840b-460801757087/blobs/sha256:315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8","statusCode":200,"latency":"167.5817ms","bodySize":104857600,"headers":{"Accept-Encoding":["gzip"],"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":565,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:18.0525589+03:00","message":"HTTP API"}
{"level":"info","module":"http","clientIP":"127.0.2.201:59979","method":"HEAD","path":"/v2/6de5f5ab-cad3-11ec-840b-460801757087/manifests/tag104857600","statusCode":200,"latency":"140.1µs","bodySize":0,"headers":{"Accept":["application/vnd.oci.image.manifest.v1+json"],"Content-Type":["application/vnd.oci.image.manifest.v1+json"],"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":565,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:18.0530308+03:00","message":"HTTP API"}
{"level":"info","module":"http","clientIP":"127.0.2.201:59979","method":"GET","path":"/v2/6de5f5ab-cad3-11ec-840b-460801757087/manifests/tag104857600","statusCode":200,"latency":"95.6µs","bodySize":392,"headers":{"Accept":["application/vnd.oci.image.manifest.v1+json"],"Accept-Encoding":["gzip"],"Content-Type":["application/vnd.oci.image.manifest.v1+json"],"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":565,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:18.0535448+03:00","message":"HTTP API"}
{"level":"debug","blob path":"/tmp/zot/6de5f5ab-cad3-11ec-840b-460801757087/blobs/sha256/bc70852f25eef4c3a4453c5772bc207059534b037689472ed8c236c65d5500a0","goroutine":565,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1260","time":"2022-05-03T14:23:18.0540366+03:00","caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1260","message":"blob path found"}
{"level":"info","module":"http","clientIP":"127.0.2.201:59979","method":"HEAD","path":"/v2/6de5f5ab-cad3-11ec-840b-460801757087/blobs/sha256:bc70852f25eef4c3a4453c5772bc207059534b037689472ed8c236c65d5500a0","statusCode":200,"latency":"89.4µs","bodySize":0,"headers":{"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":565,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:18.0540917+03:00","message":"HTTP API"}
{"level":"info","module":"http","clientIP":"127.0.2.201:59979","method":"GET","path":"/v2/6de5f5ab-cad3-11ec-840b-460801757087/blobs/sha256:bc70852f25eef4c3a4453c5772bc207059534b037689472ed8c236c65d5500a0","statusCode":200,"latency":"125.8µs","bodySize":144,"headers":{"Accept-Encoding":["gzip"],"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":565,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:18.0546877+03:00","message":"HTTP API"}
{"level":"debug","blob path":"/tmp/zot/6de5f5ab-cad3-11ec-840b-460801757087/blobs/sha256/315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8","goroutine":565,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1260","time":"2022-05-03T14:23:18.0551809+03:00","caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1260","message":"blob path found"}
{"level":"info","module":"http","clientIP":"127.0.2.201:59979","method":"HEAD","path":"/v2/6de5f5ab-cad3-11ec-840b-460801757087/blobs/sha256:315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8","statusCode":200,"latency":"96.6µs","bodySize":0,"headers":{"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":565,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:18.0552464+03:00","message":"HTTP API"}
{"level":"info","module":"http","clientIP":"127.0.0.15:56249","method":"GET","path":"/v2/6de5f967-cad3-11ec-840b-460801757087/blobs/sha256:315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8","statusCode":200,"latency":"175.5185ms","bodySize":104857600,"headers":{"Accept-Encoding":["gzip"],"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":566,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:18.0619626+03:00","message":"HTTP API"}
{"level":"info","module":"http","clientIP":"127.0.0.15:56249","method":"POST","path":"/v2/6e644c11-cad3-11ec-840b-460801757087/blobs/uploads/","statusCode":202,"latency":"1.1199ms","bodySize":0,"headers":{"Accept-Encoding":["gzip"],"Content-Length":["0"],"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":566,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:18.0636665+03:00","message":"HTTP API"}
{"level":"info","r.ContentLength":104857600,"goroutine":566,"caller":"zotregistry.io/zot/pkg/api/routes.go:1042","time":"2022-05-03T14:23:18.1519043+03:00","message":"DEBUG"}
{"level":"info","module":"http","clientIP":"127.0.2.201:59979","method":"GET","path":"/v2/6de5f5ab-cad3-11ec-840b-460801757087/blobs/sha256:315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8","statusCode":200,"latency":"132.2258ms","bodySize":104857600,"headers":{"Accept-Encoding":["gzip"],"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":565,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:18.1880042+03:00","message":"HTTP API"}
{"level":"debug","src":"/tmp/zot/6e644c11-cad3-11ec-840b-460801757087/.uploads/a175f942-72f3-4a8c-b3c9-5386dcc4eb89","dstDigest":"sha256:315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8","dst":"/tmp/zot/6e644c11-cad3-11ec-840b-460801757087/blobs/sha256/315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1139","time":"2022-05-03T14:23:18.5729845+03:00","caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1139","message":"dedupe: enter"}
{"level":"debug","blobPath":"/tmp/zot/6e644c11-cad3-11ec-840b-460801757087/blobs/sha256/315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8","dstRecord":"/tmp/zot/6aaf12f7-cad3-11ec-840b-460801757087/blobs/sha256/315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1199","time":"2022-05-03T14:23:18.5730572+03:00","caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1199","message":"dedupe: creating hard link"}
{"level":"debug","src":"/tmp/zot/6e644c11-cad3-11ec-840b-460801757087/.uploads/a175f942-72f3-4a8c-b3c9-5386dcc4eb89","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1214","time":"2022-05-03T14:23:18.5731124+03:00","caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1214","message":"dedupe: remove"}
{"level":"info","module":"http","clientIP":"127.0.0.15:56249","method":"PUT","path":"/v2/6e644c11-cad3-11ec-840b-460801757087/blobs/uploads/a175f942-72f3-4a8c-b3c9-5386dcc4eb89?digest=sha256%3A315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8","statusCode":201,"latency":"430.2393ms","bodySize":0,"headers":{"Accept-Encoding":["gzip"],"Content-Length":["104857600"],"Content-Type":["application/octet-stream"],"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":566,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:18.5821526+03:00","message":"HTTP API"}
{"level":"info","module":"http","clientIP":"127.0.0.15:56249","method":"POST","path":"/v2/6e644c11-cad3-11ec-840b-460801757087/blobs/uploads/","statusCode":202,"latency":"169.1µs","bodySize":0,"headers":{"Accept-Encoding":["gzip"],"Content-Length":["0"],"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":566,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:18.58288+03:00","message":"HTTP API"}
{"level":"info","r.ContentLength":144,"goroutine":566,"caller":"zotregistry.io/zot/pkg/api/routes.go:1042","time":"2022-05-03T14:23:18.5833892+03:00","message":"DEBUG"}
{"level":"debug","src":"/tmp/zot/6e644c11-cad3-11ec-840b-460801757087/.uploads/a039a0ab-db94-4f0b-aa5a-cbfed0754f86","dstDigest":"sha256:9907a3a440b89bfd835ae961fc4c578358981b1d85f21c2497729a599d11497d","dst":"/tmp/zot/6e644c11-cad3-11ec-840b-460801757087/blobs/sha256/9907a3a440b89bfd835ae961fc4c578358981b1d85f21c2497729a599d11497d","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1139","time":"2022-05-03T14:23:18.5835903+03:00","caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1139","message":"dedupe: enter"}
{"level":"debug","src":"/tmp/zot/6e644c11-cad3-11ec-840b-460801757087/.uploads/a039a0ab-db94-4f0b-aa5a-cbfed0754f86","dst":"/tmp/zot/6e644c11-cad3-11ec-840b-460801757087/blobs/sha256/9907a3a440b89bfd835ae961fc4c578358981b1d85f21c2497729a599d11497d","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1164","time":"2022-05-03T14:23:18.5928459+03:00","caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:1164","message":"dedupe: rename"}
{"level":"info","module":"http","clientIP":"127.0.0.15:56249","method":"PUT","path":"/v2/6e644c11-cad3-11ec-840b-460801757087/blobs/uploads/a039a0ab-db94-4f0b-aa5a-cbfed0754f86?digest=sha256%3A9907a3a440b89bfd835ae961fc4c578358981b1d85f21c2497729a599d11497d","statusCode":201,"latency":"9.5318ms","bodySize":0,"headers":{"Accept-Encoding":["gzip"],"Content-Length":["144"],"Content-Type":["application/octet-stream"],"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":566,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:18.5929197+03:00","message":"HTTP API"}
{"level":"info","blobPath":"/tmp/zot/6e644c11-cad3-11ec-840b-460801757087/blobs/sha256/315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8","reference":"tag1","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:517","time":"2022-05-03T14:23:18.5934022+03:00","caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:517","message":"manifest layers"}
{"level":"debug","digest":"sha256:f000e9f57c218c46ee94d72d6cf2c6ce333f65fab5501c0eb70dec27e81cabbf","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:144","time":"2022-05-03T14:23:18.5936146+03:00","message":"GC: got reference"}
{"level":"debug","digest":"sha256:f000e9f57c218c46ee94d72d6cf2c6ce333f65fab5501c0eb70dec27e81cabbf","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:144","time":"2022-05-03T14:23:18.5936719+03:00","message":"GC: marking from root"}
{"level":"debug","digest":"sha256:f000e9f57c218c46ee94d72d6cf2c6ce333f65fab5501c0eb70dec27e81cabbf","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:144","time":"2022-05-03T14:23:18.5937109+03:00","message":"-> ws.recurse"}
{"level":"debug","name":"github.com/opencontainers/image-spec/specs-go::Versioned","v1path":"github.com/opencontainers/image-spec/specs-go/v1","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:144","time":"2022-05-03T14:23:18.5937993+03:00","message":"detected jump outside permitted packages"}
{"level":"debug","digest":"sha256:9907a3a440b89bfd835ae961fc4c578358981b1d85f21c2497729a599d11497d","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:144","time":"2022-05-03T14:23:18.593841+03:00","message":"-> ws.recurse"}
{"level":"debug","digest":"sha256:9907a3a440b89bfd835ae961fc4c578358981b1d85f21c2497729a599d11497d","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:144","time":"2022-05-03T14:23:18.593909+03:00","message":"<- ws.recurse"}
{"level":"debug","digest":"sha256:315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:144","time":"2022-05-03T14:23:18.5939464+03:00","message":"-> ws.recurse"}
{"level":"debug","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:144","time":"2022-05-03T14:23:18.5939866+03:00","message":"skipping walk into non-parseable media-type application/vnd.oci.image.layer.v1.tar of blob sha256:315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8"}
{"level":"debug","digest":"sha256:315e899f015e25df6f9fc801965d2a3fd8c5724f22417f427d318ce074f4d1e8","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:144","time":"2022-05-03T14:23:18.5940282+03:00","message":"<- ws.recurse"}
{"level":"debug","digest":"sha256:f000e9f57c218c46ee94d72d6cf2c6ce333f65fab5501c0eb70dec27e81cabbf","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:144","time":"2022-05-03T14:23:18.5940619+03:00","message":"<- ws.recurse"}
{"level":"debug","goroutine":566,"caller":"zotregistry.io/zot/pkg/storage/storage_fs.go:144","time":"2022-05-03T14:23:18.5941372+03:00","message":"garbage collected 0 blobs"}
{"level":"info","module":"http","clientIP":"127.0.0.15:56249","method":"PUT","path":"/v2/6e644c11-cad3-11ec-840b-460801757087/manifests/tag1","statusCode":201,"latency":"873.1µs","bodySize":0,"headers":{"Accept":["application/vnd.oci.image.manifest.v1+json"],"Accept-Encoding":["gzip"],"Content-Length":["392"],"Content-Type":["application/vnd.oci.image.manifest.v1+json"],"User-Agent":["go-resty/1.12.0 (https://github.com/go-resty/resty)"]},"goroutine":566,"caller":"zotregistry.io/zot/pkg/api/session.go:132","time":"2022-05-03T14:23:18.5941701+03:00","message":"HTTP API"}
```

File: cmd/zb/perf.go [hunk 1]
```diff
@@ -754,3 +727,81 @@ func Perf(workdir, url, auth, repo string, concurrency int, requests int, outFmt
 		os.Exit(1)
 	}
 }
+
+// getRandomClientFromIPs returns a resty client with a random bind address from ips slice.
```
Comment (rchincha) [inline cmd/zb/perf.go:731]: Just call this getRandomClientIP(...)

File: .github/workflows/cluster.yaml [hunk 2]
```diff
@@ -136,7 +136,7 @@ jobs:
             ./bin/zot-linux-amd64 serve test/cluster/config-minio3.json &
             sleep 10
             # run zb
-            bin/zb-linux-amd64 -c 10 -n 50 -o ci-cd http://localhost:8080
+            bin/zb-linux-amd64 -c 10 -n 50 -o ci-cd --src-cidr 127.0.0.0/8 http://localhost:8080
```
Comment (rchincha) [inline .github/workflows/cluster.yaml:139]: We need a test for the "-i" option also, but not here.

File: cmd/zb/perf.go (hunk 1 above)
Comment (peusebiu) [reply cmd/zb/perf.go:731]: sure

File: .github/workflows/cluster.yaml (hunk 2 above)
Comment (peusebiu) [reply .github/workflows/cluster.yaml:139]: added

**Review by roxanaN — APPROVED** [review]
LGTM

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**